src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -73,58 +83,131 @@ public void storeMessage(Message message) {
 
+    Query query = createMessageQueryWithUserFilter(loggedInUser, otherUser);
+    PreparedQuery results = datastore.prepare(query);
+    return prepareQueryToMesssages(results);
+  }
+
+  private Query createMessageQueryWithUserFilter(String loggedInUser, String otherUser) {
+    return new Query("Message")
+        .setFilter(createUserFilter(loggedInUser, otherUser))
+        .addSort("timestamp", SortDirection.ASCENDING);
+  }
+
+  private Filter createUserFilter(String loggedInUser, String otherUser) {
     // Messages between two people, where logged-in user is the sender
-    Filter messagesSentByLoggedInUser = new Query.FilterPredicate("user", FilterOperator.EQUAL, loggedInUser);
-    Filter messagesReceivedByOtherUser = new Query.FilterPredicate("recipient", FilterOperator.EQUAL, otherUser);
+    Filter messagesSentByLoggedInUser =
+        new Query.FilterPredicate("user", FilterOperator.EQUAL, loggedInUser);
+    Filter messagesReceivedByOtherUser =
+        new Query.FilterPredicate("recipient", FilterOperator.EQUAL, otherUser);
 
     // All the messages sent by logged-in user to the other user
-    Filter loggedInUserMessages = CompositeFilterOperator.and(messagesSentByLoggedInUser, messagesReceivedByOtherUser);
+    Filter loggedInUserMessages =
+        CompositeFilterOperator.and(messagesSentByLoggedInUser, messagesReceivedByOtherUser);
 
     // Messages between two people, where logged-in user is the recipient
-    Filter messagesSentByOtherUser = new Query.FilterPredicate("user", FilterOperator.EQUAL, otherUser);
-    Filter messagesReceivedByLoggedInUser = new Query.FilterPredicate("recipient", FilterOperator.EQUAL, loggedInUser);
-    
+    Filter messagesSentByOtherUser =
+        new Query.FilterPredicate("user", FilterOperator.EQUAL, otherUser);
+    Filter messagesReceivedByLoggedInUser =
+        new Query.FilterPredicate("recipient", FilterOperator.EQUAL, loggedInUser);
+
     // All the messages recieved by logged-in user sent by the other user
-    Filter otherUserMessages = CompositeFilterOperator.and(messagesSentByOtherUser, messagesReceivedByLoggedInUser);
+    Filter otherUserMessages =
+        CompositeFilterOperator.and(messagesSentByOtherUser, messagesReceivedByLoggedInUser);
 
     // All the messages between the two users
-    Filter directMessages = CompositeFilterOperator.or(loggedInUserMessages, otherUserMessages);
+    return CompositeFilterOperator.or(loggedInUserMessages, otherUserMessages);
+  }
 
-    // Messages are ordered from oldest to newest since messages between
-    // people may include implicit context/reference to previously sent messages,
-    // and this order makes the chronology of the conversation and context
-    // more intuitive and understandable in the messages list page. 
-    Query query =
-        new Query("Message")
-            .setFilter(directMessages)
-            .addSort("timestamp", SortDirection.ASCENDING);
+  public List<Message> getAllMessages() {
```

Comment:
Minor comment: can you put `public` methods before `private` methods? This way, the file is more scannable in terms of which methods are available to other files.